File: python_roguelike/data/combat_action_data.py
from dataclasses import dataclass, field
from typing import Optional
from .enums import ActionType, TargetType


@dataclass
class CombatActionData:
    type: ActionType = ActionType.DealDamage
    value: int = 0
    target: TargetType = TargetType.Self
    effect_id: Optional[str] = None

File: src/Roguelike.Core/Core/Data/CombatActionData.cs
namespace Roguelike.Data
{


    public class CombatActionData
    {


        public ActionType Type { get; set; }


        public int Value { get; set; }


        public TargetType Target { get; set; }


        public string EffectId { get; set; }

        public CombatActionData() { }

        public CombatActionData(ActionType type, int value, TargetType target, string effectId = null)
        {
            Type = type;
            Value = value;
            Target = target;
            EffectId = effectId;
        }
    }
}
